lib/ts_utils/utils.py

```diff
@@ -203,7 +203,7 @@ def allowlists(distribution_name: str) -> list[str]:
 
 @functools.cache
 def get_gitignore_spec() -> pathspec.PathSpec:
-    with open(".gitignore", encoding="UTF-8") as f:
+    with Path(".gitignore").open(encoding="UTF-8") as f:
```

Comment:
I prefer the simple `open()` in this and similar cases. But it's no deal breaker for me. The rest looks good.